Library/brew.rb

```diff
@@ -4,16 +4,17 @@
 
 HOMEBREW_BREW_FILE = ENV['HOMEBREW_BREW_FILE']
 
-if ARGV == %w{--prefix}
-  puts File.dirname(File.dirname(HOMEBREW_BREW_FILE))
-  exit 0
-end
-
 require 'pathname'
 HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent.join("Homebrew")
 $:.unshift(HOMEBREW_LIBRARY_PATH.to_s)
+$:.unshift("#{HOMEBREW_LIBRARY_PATH}/test/lib") if ENV["HOMEBREW_TEST_LIB"]
```

Comment:
It's no longer really an integration test at that point, unfortunately, because it's no longer calling the `HOMEBREW_PREFIX/bin/brew` script. I'm definitely open to other approaches here, though.